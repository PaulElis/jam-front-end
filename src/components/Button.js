import React, { Component } from 'react';
import '../styles/Button.css'

class Button extends Component {
  render() {
    // console.log('props', this.props);
    return (
        <a href={this.props.link} target="_blank" rel="noopener noreferrer" id='button'>
          {this.props.children} </a>
    );
  }
}

export default Button;
