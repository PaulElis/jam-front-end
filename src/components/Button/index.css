a#button{
  color: #000;
  font-size: 4vw;
  font-weight: bold;
  letter-spacing: 2px;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 5px;
  width: 50%;
  padding: 1% 32%;
  text-decoration: none;
  margin-bottom: 5%;
}

a#button:hover{
  transition: 1s;
  background-color: #eee;
}

@media only screen and (min-width: 576px) {
a#button{
  font-size: 1vw;
  padding: 1% 38%;
}
}

@media only screen and (min-width: 768px) {
a#button{
  font-size: 1vw;
  padding: 1% 35%;
}
}

@media only screen and (min-width: 768px) {
a#button{
  font-size: .75vw;
  padding: 1% 38%;
}
}
