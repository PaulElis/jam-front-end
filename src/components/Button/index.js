import React, { Component } from 'react';
import './index.css'

class Button extends Component {

  renderCSSTag = () => {
    return this.props.top_artists === undefined && this.props.artist ?
      'favoriteartist-' : this.props.top_artists === undefined ?
      'favoritealbum-' : ''
  }

  render() {
    // console.log('props', this.props);
    return (
      <a href={this.props.link}
        target="_blank"
        rel="noopener noreferrer"
        id={`${this.renderCSSTag()}button`}
        link={this.props.artist ? this.props.artist.url : this.props.album.url}
      >
        {this.props.artist ? 'View Artist' : 'View Album'}
      </a>
    );
  }
}

export default Button;
