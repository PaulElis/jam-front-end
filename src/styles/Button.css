a#button{
  color: #000;
  font-size: 4vw;
  font-weight: bold;
  background-color: #fff;
  border: 1px solid #ddd;
  border-radius: 5px;
  padding: 1%;
  text-decoration: none;
}

a#button:hover{
  transition: 1s;
  background-color: #ddd;
}
