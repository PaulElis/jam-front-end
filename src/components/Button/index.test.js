const index = require('./index');

import '../../setupEnzyme'
import { shallow } from 'enzyme';
import React from 'react';
import Button from './index';


describe('Button', () => {
  test('snapshot Button component', () => {
    expect(shallow(<Button />)).toMatchSnapshot();
  })
})
