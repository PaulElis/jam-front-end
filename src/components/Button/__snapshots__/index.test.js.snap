// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot Button component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        undefined,
        " ",
      ],
      "href": undefined,
      "id": "favoritealbum-button",
      "rel": "noopener noreferrer",
      "target": "_blank",
    },
    "ref": null,
    "rendered": Array [
      undefined,
      " ",
    ],
    "type": "a",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          undefined,
          " ",
        ],
        "href": undefined,
        "id": "favoritealbum-button",
        "rel": "noopener noreferrer",
        "target": "_blank",
      },
      "ref": null,
      "rendered": Array [
        undefined,
        " ",
      ],
      "type": "a",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
